webapp/src/components/systemConsole/surveyDateTime/surveyDateTime.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {format, parse} from 'date-fns';
+
+import SurveyTimeSelector from 'components/surveyTimeSelector/surveyTimeSelector';
+import SurveyDateSelector from 'components/surveyDateSelector/surveyDateSelector';
+import {CustomComponentProps} from 'types/mattermost-webapp';
+
+import './style.scss';
+
+const defaultSurveyTime = '09:00';
+
+// date 30 days from today
+const defaultSurveyDate = new Date(new Date().setDate((new Date()).getDate() + 30));
+
+function SurveyDateTime({id, setSaveNeeded, onChange, config}: CustomComponentProps) {
+    const [surveyTime, setSurveyTime] = useState<string>(defaultSurveyTime);
+    const [surveyDate, setSurveyDate] = useState<Date>(defaultSurveyDate);
+
+    useEffect(() => {
+        // sets the date picker and time picker to the values saved in config on load
+
+        const dateTimeConfig = config.PluginSettings?.Plugins?.['com.mattermost.user-survey']?.surveydatetime;
+
+        if (dateTimeConfig?.time) {
+            setSurveyTime(dateTimeConfig.time);
```

Comment:
Why do we set a state in an effect (creating an extra render) instead of using the value from the props directly on render?

If it is about reducing the amount of times we parse the time string, we can use a useMemo instead.

Reply:
Over here we're setting the value from props into the state. A value may not always exist in the props, for example in case of first time installing the plugin. So we need to fallback to a default value (`defaultSurveyTime`). So I start with a state populated by the default value, then use useEffect to check if a value is present in props. The logic is lengthy enough to not be put in useState(), thats why.